.github/workflows/build-circuits.yml

```diff
@@ -183,6 +183,21 @@ jobs:
       - name: Replace Prover Makefile # TODO: Make a fork generate the appropriate Linux Makefile
         run: cp .github/resources/prover/Makefile rapidsnark/Makefile
 
+      - name: Cache GMP Archive
+        id: cache-gmp
+        uses: actions/cache@v4
+        with:
+          path: rapidsnark/depends/gmp-6.2.1.tar.xz
+          key: gmp-6.2.1
+
+      - name: Download GMP Archive
+        if: steps.cache-gmp.outputs.cache-hit != 'true'
+        working-directory: rapidsnark/depends
+        run: |
+          echo "Downloading GMP archive..."
+          curl -L -o gmp-6.2.1.tar.xz https://ftpmirror.gnu.org/gmp/gmp-6.2.1.tar.xz
+          echo "Download complete."
```

Comment:
The normal `build_gmp.sh` script is using `https://ftp.gnu.org/gmp/gmp-6.2.1.tar.xz` to download the archive. The main gnu ftp server is not responding right now. `ftpmirror.gnu.org` is the recommended way to download from gnu ftp anyways, it will find the closest trusted mirror near you.